code/datums/components/nanites.dm

```diff
@@ -345,27 +345,33 @@
 /datum/component/nanites/proc/nanite_scan(datum/source, mob/user, full_scan)
 	SIGNAL_HANDLER
 
+	var/list/message = list()
+
 	if(!full_scan)
+		. = TRUE
```

Comment:
return value shouldn't be set to `TRUE` this early since this is before the `stealth` check. move this `. = TRUE` to where the return was previously (after the stealth check)